### PR TITLE
MGDCTRS-1727: Propagate metadata information as metrics tag to fleetshard components

### PR DESCRIPTION
**Pull request by lburgazzoli**
There is some more work to be done to cleanup the implementation in particular about how metrics are configured so I'd like to include some of the changes in that area included in https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/pull/401 but as as follow up to reduce the amount of changes.

### CODE REVIEW

**Review by lgarciaaco — COMMENTED** [review]
We can transform pod labels into prometheus labels. Isn't that what this PR is doing?

https://github.com/bf2fc6cc711aee1a0c2a/cos-observability-resources/blob/main/resources/prometheus/pod_monitors/cos-fleetshard-connectors-camel.yaml#L19-L43

**Comment by lburgazzoli** [comment]
@lgarciaaco I removed the camel specific bits and I'll amend the observability resources

**Comment by lgarciaaco** [comment]
@lburgazzoli I think on top of https://github.com/bf2fc6cc711aee1a0c2a/cos-observability-resources/pull/7 you might want to add the labels for `org_id` and `price_tier` to metrix exposed by the sync, specifically `cos_fleetshard_sync_connector_state` and `cos_fleetshard_sync_connector_state_count_total` metrics.

https://github.com/bf2fc6cc711aee1a0c2a/cos-observability-resources/pull/7 add the labels to the connector pod related metrics(cpu, load, mem) and camel related metrics exposed from that pod (exchanges...) but we rarely use those metrics. We heavily work with `cos_fleetshard_sync_connector_state_count_total` and `cos_fleetshard_sync_connector_state` so I think we might want them there too.

**Comment by lburgazzoli** [comment]
> @lburgazzoli I think on top of [bf2fc6cc711aee1a0c2a/cos-observability-resources#7](https://github.com/bf2fc6cc711aee1a0c2a/cos-observability-resources/pull/7) you might want to add the labels for `org_id` and `price_tier` to metrix exposed by the sync, specifically `cos_fleetshard_sync_connector_state` and `cos_fleetshard_sync_connector_state_count_total` metrics.
> 

Is there any specific YAML in the cos-observability-resources where it should be added ? As today, the sync adds such information to every connector [status update](https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/pull/545/files#diff-847053de838b6d0fe5229098b8ae9e77dadf2fa79e8079d0af0341fc6844d0e7R100-R104) which is tested [here](https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/pull/545/files#diff-5a7b0af07cd0c5729689ce6bb9b88e3cf34d9d2fff9cf4fa22b80a782f0a08eeR120-R158)

**Comment by lgarciaaco** [comment]
> > @lburgazzoli I think on top of [bf2fc6cc711aee1a0c2a/cos-observability-resources#7](https://github.com/bf2fc6cc711aee1a0c2a/cos-observability-resources/pull/7) you might want to add the labels for `org_id` and `price_tier` to metrix exposed by the sync, specifically `cos_fleetshard_sync_connector_state` and `cos_fleetshard_sync_connector_state_count_total` metrics.
> 
> Is there any specific YAML in the cos-observability-resources where it should be added ? As today, the sync adds such information to every connector [status update](https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/pull/545/files#diff-847053de838b6d0fe5229098b8ae9e77dadf2fa79e8079d0af0341fc6844d0e7R100-R104) which is tested [here](https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/pull/545/files#diff-5a7b0af07cd0c5729689ce6bb9b88e3cf34d9d2fff9cf4fa22b80a782f0a08eeR120-R158)

You good to go then, I just didn't read through all the PR considering its size :-)

**Review by lgarciaaco — APPROVED** [review]

File: etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/kubernetes.yml [hunk 1]
```diff
@@ -410,10 +410,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: "metadata.namespace"
-        - name: "SMALLRYE_CONFIG_LOCATIONS"
-          value: "/mnt/app-config/configuration"
         - name: "SMALLRYE_CONFIG_SOURCE_FILE_LOCATIONS"
           value: "/mnt/app-secret"
+        - name: "SMALLRYE_CONFIG_LOCATIONS"
```
Comment (rk3rn3r) [inline etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/kubernetes.yml:415]: This line moves all the time with every build. maybe, to make the PR smaller, we should not change this file?

**Review by rk3rn3r — APPROVED** [review]

File: etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/kubernetes.yml (hunk 1 above)
Comment (lburgazzoli) [reply etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/kubernetes.yml:415]: There is a fix for that on another PR

File: cos-fleetshard-operator-camel/src/main/java/org/bf2/cos/fleetshard/operator/camel/CamelOperandSupport.java [hunk 2]
```diff
@@ -334,6 +346,28 @@ public static Map<String, String> createSecretsData(
...
+
+        if (!defaultTags.isEmpty()) {
+            //
+            // For camel based connector, metrics are exposed using the mp-metrics component/extension.
+            // According to the mp-metrics specs, tags can be supplied in two ways:
+            //
+            // 1. Programmatically
+            // 2. By using MicroProfile Config and setting a configuration property of the
+            //    name mp.metrics.tags. Tag values set through mp.metrics.tags MUST escape
+            //    equal symbols = and commas , with a backslash \
```
Comment (rinaldodev) [inline cos-fleetshard-operator-camel/src/main/java/org/bf2/cos/fleetshard/operator/camel/CamelOperandSupport.java:363]: Was this code expected to do the escaping?

File: etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/crds/managedconnectors.cos.bf2.org-v1.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ spec:
     - jsonPath: .spec.connectorId
       name: CONNECTOR_ID
       type: string
-    - jsonPath: .spec.deployment.connectorTypeId
+    - jsonPath: .status.deployment.connectorTypeId
```
Comment (rinaldodev) [inline etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/crds/managedconnectors.cos.bf2.org-v1.yml:22]: I believe this is changing to the wrong version?

File: cos-fleetshard-operator-camel/src/main/java/org/bf2/cos/fleetshard/operator/camel/CamelOperandSupport.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lburgazzoli) [reply cos-fleetshard-operator-camel/src/main/java/org/bf2/cos/fleetshard/operator/camel/CamelOperandSupport.java:363]: no it was not

File: etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/crds/managedconnectors.cos.bf2.org-v1.yml (hunk 3 above)
Comment (lburgazzoli) [reply etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/crds/managedconnectors.cos.bf2.org-v1.yml:22]: I must rebuild to get the resource sanitization reset it

**Review by rinaldodev — APPROVED** [review]

**Pull request merged**